src/decoders/novatel/src/rangecmp/range_decompressor.cpp

```diff
@@ -868,8 +884,11 @@ void RangeDecompressor::RangeCmp4ToRange(uint8_t* pucCompressedData_, RangeStruc
                             else { DecompressReferenceBlock<true>(&pucTempDataPointer, stMeasurementBlock, eMeasurementSource); }
 
                             stChannelTrackingStatus = ChannelTrackingStatusStruct(eCurrentSatelliteSystem, eCurrentSignalType, stMeasurementBlock);
-                            PopulateNextRangeData((stRangeMessage_.astRangeData[stRangeMessage_.uiNumberOfObservations++]), stMeasurementBlock,
-                                                  stMetaData_, stChannelTrackingStatus, uiPRN, stMeasurementBlockHeader.cGLONASSFrequencyNumber);
+                            obsStatus = PopulateNextRangeData((stRangeMessage_.astRangeData[stRangeMessage_.uiNumberOfObservations]),
+                                                              stMeasurementBlock, stMetaData_, stChannelTrackingStatus, uiPRN,
+                                                              stMeasurementBlockHeader.cGLONASSFrequencyNumber, bAllowUnknownObs);
+
+                            if (obsStatus) { stRangeMessage_.uiNumberOfObservations++; }
```

Comment:
You should be comparing against the enum type. This code could lead to bugs otherwise.

Reply:
sounds good!